GoogleChromeLabs/ps-analysis-tool
# Feature: Expand support for GDPR banner acceptance in CLI

## Description
This PR allows the user to pass a `selector.json` file with flag `-b` or `--button-selectors` which will be used when any custom selector is used for GDPR banner acceptance.
It also adds support for major CMP's like OneTrust, Ibuenda etc.

## Relevant Technical Choices
- Use constant array for selector CSS.
- Use constant array for Text selector.
- Add a custom function for using selector.json provided by the user.

## Testing Instructions
- Clone this branch.
- In the terminal run `npm i && npm run build:cli:dashboard`.
- Then run the CLI on site doctolib.fr using `npm run cli https://doctolib.fr -- -b selector.json`.
- Add the following in selector.json.
```
{
    "cssSelectors":["didomi-notice-agree-button"],
    "textSelectors": [],
    "xPath": []
}
```

## Additional Information:
- If the user provided selectors aren't found in the page then the CLI falls back to using major selector, if that fails then we use the text selectors in the language.

## Screenshot/Screencast
![Untitled-2023-06-16-1319](https://github.com/user-attachments/assets/61dc46c5-c980-48ea-9e3d-7983877744a1)


---

## Checklist






Fixes #810
